### DISCUSSION TITLE
training with labels in same  : detetcion only then work a the same place

### DISCUSSION THREAD

**Discussion by veevee2** [ultralytics/ultralytics#16398] [Q&A]
hi,
i train with labels (180000) around the same place on each images.
The detection result are good but only around the same location of the tranings labels.
Other elements outside this area are not detected 

Is there any reason the detection only work around the same place of the traning labels ?
Any tuning to solve that ?

I looked for augmentation (with Albumentations) to create labels in différents place, but not found any chaning the location of labels.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @veevee2, thank you for reaching out to Ultralytics 🚀! We recommend checking out the [Docs](https://docs.ultralytics.com) where you can find a wealth of information and examples, including guidance on using [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/).

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) so we can better assist in debugging the issue.

If this relates to custom training ❓, it would be helpful to include additional details such as dataset examples and training logs. Make sure you're following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

You're also welcome to join our Ultralytics community for more interactive support! Join our [Discord](https://ultralytics.com/discord) 🎧 for real-time discussions, share insights on our [Subreddit](https://reddit.com/r/ultralytics), or dive into more detailed conversations on [Discourse](https://community.ultralytics.com).

## Upgrade

To ensure you're using the latest features and fixes, upgrade to the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLOv8 can run smoothly in the following verified environments:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud**: Use Deep Learning VMs, see the [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon**: Use Deep Learning AMIs, see the [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker**: Check out our [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, our [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are all passing, confirming correct operation of all YOLOv8 [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across platforms.

Please note that this is an automated response, and an Ultralytics engineer will assist you soon. 😊

**Comment by glenn-jocher** [comment]
It sounds like your model is overfitting to the specific locations in your training data. Consider using data augmentation techniques to vary the position of objects in your images. You might want to explore random cropping or rotation to help diversify the training data.

**Comment by veevee2** [comment]
thanks,
so i tried differents augmentations
- a.crop : augmentation properly applied on my 10k training images but worst result, no detection after training
- a.RandomGridShuffle : fall on a 'problem' for now as image zones are properly shuffled, but not the labels ( question opened here :  ([RandomGridShuffle ](https://github.com/albumentations-team/albumentations/issues/1948) )
- will try rotation

**Reply by glenn-jocher** [reply]
@veevee2 it sounds like your model is overfitting to specific areas. Consider using augmentations that change the spatial distribution of objects, such as rotation or translation, to help generalize the model. If you encounter issues with specific augmentations, checking the Albumentations documentation or community might provide solutions.
